client/src/components/AutoHeightVirtualList.js

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { List } from "react-virtualized";
+
+export class AutoHeightVirtualList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.max_height = props.max_height;
+    this.list_ref = this.props.list_ref || React.createRef();
+
+    this.state = {
+      list_height: this.max_height,
+    };
+  }
+
+  componentDidUpdate(prev_props, prev_state) {
+    const { max_height } = this.props;
+
+    if (max_height !== this.max_height) {
+      this.max_height = max_height;
```

Comment:
I don't see where `this.max_height` is necessary, am I pissing something? Can't you just always use the prop value?